esp/src/eclwatch/WUQueryWidget.js

```diff
@@ -516,13 +516,14 @@ define([
                         context.downloadToList.set("disabled", true);
                     }
                 });
-                ESPUtil.goToPageUserPreference(this.workunitsGrid, "WUQueryWidget");
-                aspect.after(this.workunitsGrid, "gotoPage", function (deferred, args) {
-                    return deferred.then(function () {
-                        args[0] > 1 ? context._idleWatcher.stop() : context._idleWatcher.start();
+                ESPUtil.goToPageUserPreference(this.workunitsGrid, "WUQueryWidget_GridRowsPerPage").then(function(){
+                    aspect.after(context.workunitsGrid, "gotoPage", function (deferred, args) {
+                        return deferred.then(function () {
+                            args[0] > 1 ? context._idleWatcher.stop() : context._idleWatcher.start();
+                        });
                     });
```

Comment:
There is no need to put the "aspect.after" calls in the "then" - its that startup thats important.